source/flow.js

```diff
@@ -0,0 +1,37 @@
+import applyTo from './applyTo.js';
+import _curry2 from './internal/_curry2.js';
+
+/**
+ * Takes the value of an expression and applies it to a function
+ * which is the left-to-right serial composition of the functions
+ * given in the second argument.
+ *
+ * The functions in the pipeline should be unary functions.
+ *
+ * `flow` is helps to avoid introducing named functions with named arguments
+ * for computing the result of a function pipeline which depends on given initial values.
+ * Rather than composing a custom pipeline function `p = (_x, _y) => R.pipe(g(x), h(y), …)`
+ * which is only later needed once `z = p(x, y)`,
+ * the introduction of `p`, `_x` and `_y` can be avoided: `z = flow(x, [g, h(y),…]`
+ *
+ * In some libraries this function is named `pipe`.
+ *
+ * @func
+ * @memberOf R
+ * @since v0.30.0
+ * @category Function
+ * @sig a → [(a → b), …, (y → z)] → z
+ * @param {*} a The seed value
+ * @param {Array<Function>} pipeline functions composing the pipeline
+ * @return {*} z The result of applying the seed value to the function pipeline
+ * @see R.pipe
+ * @example
+ *      const defaultName = 'Jane Doe';
+ *      const savedName = R.flow(localStorage.get('name'), [R.when(R.isNil(defaultName)), R.match(/(.+)\s/), R.nth(0)]);
+ *      const givenName = R.flow($givenName.value, [R.trim, R.when(R.isEmpty, R.always(savedName))])
```

Comment:
I think this is far too complex for an example function.

Can we just use something more like the test cases?:

```js
 R.flow(9, [Math.sqrt, R.negate, R.inc]), //=> -2
```

or 

```js
const myFn = R.flow(9, [Math.sqrt, R.multiply, R.map])
myFn ([1, 2, 3]) //> [3, 6, 9]
```

Reply:
As a non FP or Ramda expert, I find examples beyond “hello world” valuable and useful to give me ideas on how to use and compose/combine stuff together.

For example, let’s consider documentation for `identity` like this:

```javascript
identity(1);
//=> 1
```

It doesn’t help me understand or get ideas for more real situation where I could use it.

On the other hand, I understand it is not any library’s job to teach people how to program. The examples in JSDoc (which I believe are extracted to be turned in the documentation website) must be simple and concise, I agree, and they can’t possible cover all use cases anyway.

Still, it could be useful to place those more complex and/or real life examples in some other place, maybe in the [Cookbook](https://github.com/ramda/ramda/wiki/Cookbook) instead of just throwing them away.

I’m not sure if this idea makes much too much sense, but as a person learning FP and Ramda, I thought I’d mention it anyway.

Reply:
> It doesn’t help me understand or get ideas for more real situation where I could use it.

Indeed, and that is exactly why I've attempted a more real example.

For me the cookbook is seldom the solution and only in desperate cases. It is far too wide-stretched and good for learning, but bad for look-up. (Before filtering the correct solution from the vast examples of the cookbook one could probably go faster using imperative-style mix-in JS)

Btw. another API example which leaves me confused is `R.ap` with arrays.